velox/common/memory/MemoryArbitrator.h

```diff
@@ -312,10 +321,18 @@ class MemoryReclaimer {
   /// Invoked by the memory arbitrator to reclaim from memory 'pool' with
   /// specified 'targetBytes'. It is expected to reclaim at least that amount of
   /// memory bytes but there is no guarantees. If 'targetBytes' is zero, then it
-  /// reclaims all the reclaimable memory from the memory 'pool'. The function
-  /// returns the actual reclaimed memory bytes.
-  virtual uint64_t
-  reclaim(MemoryPool* pool, uint64_t targetBytes, Stats& stats);
+  /// reclaims all the reclaimable memory from the memory 'pool'. 'maxWaitMs'
+  /// specifies the max time to wait for reclaim if not zero. The memory
+  /// reclaim might fail if exceeds the timeout. The function returns the actual
+  /// reclaimed memory bytes.
+  ///
+  /// NOTE: 'maxWaitMs' is optional and the actual memory reclaim implementation
```

Comment:
Would it make sense to make this required and expect that all implementation do respect the timeout?

Reply:
> Let's first respect timeout in task pause and consider to cover the other use cases if we run into similar cluster wide problem? WDYT?

This is usually problematic because if don't have safeguards in place we may discover a problem in production and that will be cause a SEV. Investigating hairy issues like this in the middle of the night is never fun and scrambling for a quick patch to unblock production is also not very enjoyable.

Reply:
Make sense. I will think about how to enforce that next.